hpcgap/lib/mapping.gi

```diff
@@ -276,16 +302,18 @@ InstallGlobalFunction( PreImage, function ( arg )
 
         # preimage of a single element <img> under <map>
         if     FamRangeEqFamElm( FamilyObj( map ), FamilyObj( img ) ) then
-          if not (     IsGeneralMapping( map ) and IsInjective( map )
-                   and IsSurjective( map ) ) then
-            Error( "<map> must be an inj. and surj. mapping" );
-          elif img in Range( map ) then
-            return PreImageElm( map, img );
+          if not ( IsInjective( map ) and IsSurjective( map ) ) then
+            ErrorNoReturn( "<map> must be an inj. and surj. mapping" );
```

Comment:
I know this isn't your message, but I recommend removing the abbreviations, to make things as clear as possible.
```suggestion
            ErrorNoReturn( "<map> must be an injective and surjective mapping" );
```

Reply:
Actually, why not just say "bijective" (and use `IsBijective`)?

Reply:
As well as `IsInjective` and `IsSurjective`, `IsBijective` I implies `IsTotal` and `IsSingleValued`. I don’t know, but perhaps there’s a reason we don’t want to require one or the other of those things here

Reply:
Ugh, yeah, a map from a non-trivial group to a trivial group which is only defined at `1` is an injective and surjective mapping, even single valued, but it is not total. (That's -1 points for GAP for making a very surprising change in the meaning of very well established mathematical terms, with surprising consequences *sigh*).

Reply:
Done.